quantum/rgblight.h

```diff
@@ -219,16 +230,30 @@ typedef union {
     };
 } rgblight_config_t;
 
+// Need to maintain the old bit structure for the default case
+#if RGBLIGHT_MAX_LAYERS <= 8
 typedef struct _rgblight_status_t {
     uint8_t base_mode;
     bool    timer_enabled;
 #    ifdef RGBLIGHT_SPLIT
     uint8_t change_flags;
 #    endif
 #    ifdef RGBLIGHT_LAYERS
-    uint8_t enabled_layer_mask;
+    rgblight_layer_mask_t enabled_layer_mask;
 #    endif
 } rgblight_status_t;
+#else
+typedef struct PACKED _rgblight_status_t {
```

Comment:
My only concern here is alignment. In the old code the compiler got to pick the alignment and size of the struct, but now it's being specified explicitly. Will there be any issues if the user sets `RGBLIGHT_LAYERS = 9` or something weird?

Reply:
I did think of this. `enabled_layer_mask` and `timer_enabled` are still byte-aligned, regardless of `RGBLIGHT_MAX_LAYERS` value. The only slight worry is that `change_flags` is no longer byte aligned. I did look into the implications of that, and I don't think they will cause any problems. The main issue with a non-byte aligned struct member is that one cannot get a pointer to that _individual_ member of the struct. Looking at the entire codebase, I do not see any place where we are using such a construct, so I think we are OK.

That being said, I do really hope that we can get some thorough testing of the >8 layers case on a split board before this PR is merged.